includes/class-dispatcher.php

```diff
@@ -0,0 +1,206 @@
+	/**
+	 * Initialize the class, registering WordPress hooks.
+	 */
+	public static function init() {
+		\add_action( 'activitypub_process_outbox', array( self::class, 'process_outbox' ), 10, 1 );
+
+		// Default filters to add Inboxes to sent to.
+		\add_filter( 'activitypub_send_to_inboxes', array( self::class, 'add_inboxes_of_follower' ), 10, 2 );
+		\add_filter( 'activitypub_send_to_inboxes', array( self::class, 'add_inboxes_by_mentioned_actors' ), 10, 3 );
+		\add_filter( 'activitypub_send_to_inboxes', array( self::class, 'add_inboxes_of_replied_urls' ), 10, 3 );
+	}
+
+	/**
+	 * Process the outbox.
+	 *
+	 * @param int $id The outbox ID.
+	 */
+	public static function process_outbox( $id ) {
+		$outbox_item = \get_post( $id );
+
+		// If the activity is not a post, return.
+		if ( ! $outbox_item ) {
+			return;
+		}
+
+		$actor_type = \get_post_meta( $outbox_item->ID, '_activitypub_activity_actor', true );
+
+		switch ( $actor_type ) {
+			case 'blog':
+				$actor_id = Actors::BLOG_USER_ID;
+				break;
+			case 'application':
+				$actor_id = Actors::APPLICATION_USER_ID;
+				break;
+			case 'user':
+			default:
+				$actor_id = $outbox_item->post_author;
+				break;
+		}
+
+		$type        = \get_post_meta( $outbox_item->ID, '_activitypub_activity_type', true );
+		$transformer = Factory::get_transformer( $outbox_item->post_content );
+		$activity    = $transformer->to_activity( $type );
+
+		self::send_activity_to_followers( $activity, $actor_id, $outbox_item );
+	}
+
+	/**
+	 * Send an Activity to all followers and mentioned users.
+	 *
+	 * @param Activity $activity  The ActivityPub Activity.
+	 * @param int      $actor_id  The actor ID.
+	 * @param \WP_Post $outbox_item The WordPress object.
+	 */
+	private static function send_activity_to_followers( $activity, $actor_id, $outbox_item = null ) {
+		/**
+		 * Filters whether to send an Activity to followers.
+		 *
+		 * @param bool     $send_activity_to_followers Whether to send the Activity to followers.
+		 * @param Activity $activity                   The ActivityPub Activity.
+		 * @param int      $actor_id                   The actor ID.
+		 * @param \WP_Post $outbox_item                The WordPress object.
+		 */
+		if ( ! apply_filters( 'activitypub_send_activity_to_followers', true, $activity, $actor_id, $outbox_item ) ) {
+			return;
+		}
+
+		/**
+		 * Filters the list of inboxes to send the Activity to.
+		 *
+		 * @param array    $inboxes  The list of inboxes to send to.
+		 * @param int      $actor_id The actor ID.
+		 * @param Activity $activity The ActivityPub Activity.
+		 */
+		$inboxes = apply_filters( 'activitypub_send_to_inboxes', array(), $actor_id, $activity );
+		$inboxes = array_unique( $inboxes );
+
+		if ( empty( $inboxes ) ) {
+			return;
+		}
+
+		$json = $activity->to_json();
+
+		foreach ( $inboxes as $inbox ) {
+			safe_remote_post( $inbox, $json, $actor_id );
+		}
+
+		$outbox_item->post_status = 'publish';
+		\wp_update_post( $outbox_item );
```

Comment:
```suggestion
		\wp_publish_post( $outbox_item );
```
I don't know if one is "better" than the other but I wanted to offer it